faw/pdf-observatory/common/processor.ts

```diff
@@ -143,87 +160,115 @@ export function reprocess(decisionDefinition: DslResult, reprocessPdfGroups: Pdf
           }
 
           console.log(check);
-          throw new Error(`Check type ${check.type}`);
+          throw new Error(
+            `Could not evaluate check for ${identifier}: Unrecognized check type ${check.type}`
+          );
         };
 
         const r = evalInner(check);
         return new Set(r.map((x, i) => [x, i]).filter(x => !!x[0]).map(x => x[1]));
       };
 
-      for (const r of rs) {
-        if (r.pat.test(k)) {
+      for (const r of rules) {
+        if (r.pat.test(message)) {
           // Do we need to constrain the matching set based on an auxiliary
           // check?
           if (r.check !== null) {
-            const group = evalCheck(k, r.check);
-            if (group.size > 0) filesSubset = filesSubset.filter(x => group.has(x[0]));
-            else continue;
-          }
-          matched = true;
-          break;
-        }
-      }
-
-      if (f.all) {
-        // If we're matching all, then we want to note the set of PDFs for
-        // which any message did not match.
-        if (!matched) {
-          for (const file of filesSubset) {
-            result.add(file[0]);
-            pdfMap.get(file[0])!.info.push(`'${f.name}' rejected '${k}'`);
+            const group = evalCheck(message, r.check);
+            if (group.size > 0) {
+              // Merge `group` into `filesSubset`
+              group.forEach(filesSubset.add, filesSubset);
+            }
+          } else {
```

Comment:
The other formatting had less indentation. I think that, given your other refactors, flipping this would make sense -- do a `r.check === null` check, and in the body of that conditional, perform the short circuit and break. Then, the for loop body continues at the lower indentation level.